core/src/test/java/com/datastax/oss/driver/internal/core/cql/PoolBehavior.java

```diff
@@ -59,7 +58,7 @@ public PoolBehavior(Node node, boolean createChannel) {
       this.channel = mock(DriverChannel.class);
       EventLoop eventLoop = mock(EventLoop.class);
       ChannelConfig config = mock(DefaultSocketChannelConfig.class);
-      this.writePromise = GlobalEventExecutor.INSTANCE.newPromise();
```

Comment:
what was the reason why `GlobalEventExecutor` was not performing well here?
Was it because: 
```
and stops it when there is no
 * task pending in the task queue for 1 second`
```
?

Reply:
No it's because it's a true event loop that runs all its tasks asynchronously in a separate thread. We want to run such tasks synchronously to make tests even more deterministic than they currently are.